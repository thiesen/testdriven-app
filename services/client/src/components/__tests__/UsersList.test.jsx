import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    active: true,
    email: 'thiesen@example.org',
    id: 1,
    username: 'thiesen',
  },
  {
    active: false,
    email: 'xunda@example.org',
    id: 2,
    username: 'xunda',
  },
];

describe('<UsersList />', () => {
  it('renders given user list', () => {
    const wrapper = shallow(<UsersList users={users}/>);

    const element = wrapper.find('h4');

    expect(element.length).toBe(2);
    expect(element.get(0).props.className).toBe('well');
    expect(element.get(0).props.children).toBe('thiesen');
    expect(element.get(1).props.children).toBe('xunda');
  });

  test('snapshot', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
