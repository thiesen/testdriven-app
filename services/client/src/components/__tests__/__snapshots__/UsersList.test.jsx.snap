// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> snapshot 1`] = `
<div>
  <h4
    className="well"
  >
    thiesen
  </h4>
  <h4
    className="well"
  >
    xunda
  </h4>
</div>
`;
